src/sonarqube-gui-plugin/src/main/resources/static/js/cloneViewer/CloneClassSelector.js

```diff
@@ -0,0 +1,139 @@
+    });
+    this.renderAll();
+  };
+
+  /**
+   * Generates HTML of dependant of the current state of the object.
+   * Then overwrites the HTML in the visible webpage as well.
+   * @return {void}
+   */
+  this.renderAll = function(){
+    this.elem.html("");
+
+    // put together the selectors html string
+    var html = [];
+    html.push('<div>');
+    html.push('<select id="cloneClassSelector">')
+    this.cloneClassList.forEach(function(cloneClass, i){
+        html.push("<option value="+i+"> "+cloneClass.name+"</option>");
+    });
+    html.push('</select>');
+    html.push('</div>');
+    html.push('<div id="cloneClassMetricsContainer"></div>');
+
+    this.elem.append(html.join(""));
+
+    this.renderMetrics();
+
+    // make selectmenu
+    $("#cloneClassSelector").selectmenu();
+    $("#cloneClassSelector").val(this.selected).selectmenu("refresh");
+
+  };
+
+  this.renderMetrics = function(){
+    var div = $('#cloneClassMetricsContainer');
+    var html = [];
+    var data = this.cloneClassList[this.selected].cloneClassMetrics;
+    var metrics = SM.state[SM.options.component.key].clone.classMetrics;
+    metrics.forEach(function(metric) {
+      html.push(SM.cloneViewer.getFormatedMetric(data[metric.title], metric));
+    });
+    div.html(html.join(""));
+    $("#cloneClassSelectorContainer .sm-cloneviewer-metric-title-container").tooltip({
+      content: function() {
+        return $(this).prop('title');
+      }
+    });
+  };
+
+  /**
+   * selects a CloneClass, and calls everything that needs to update its state.
+   * @param  {int} choice    the id of the cloneclass in the this.CloneClassList
+   * @return {void}
+   */
+  this.select = function (choice){
+    this.selected = choice;
+    this.renderMetrics();
+    this.emit("onSelect", choice);
+  };
+
+  this.onCloneClassChange = function(event, ui){
+    this.select(ui.item.index);
+  };
+
+  this.bindElement = function(elem) {
+    if (this.elem !== null && jQuery.contains(document, this.elem[0])) {
+      this.elem.html("");
+    }
+    this.elem = elem;
+    if (this.elem !== null && jQuery.contains(document, this.elem[0])) {
+      this.renderAll();
+      this.registerEvents();
+    }
+  };
+
+  this.registerEvents = function() {
+    this.elem.on("selectmenuselect", "#cloneClassSelector", this.onCloneClassChange);
+  };
+
+  this.getFormatedMetric = function(val, metric) {
+    if (val === undefined) {
+      return '-';
+    }
+    SM.state[SM.options.component.key].clone.classMetrics.forEach(function(tempMetric){
+      if (metric === tempMetric.title){
+        metric = tempMetric;
+      }
+    });
+    metric = metric || { direction: 0 };
+    var greenClass = "sm-widget-threshold-green";
+    var redClass = "sm-widget-threshold-red";
+    var faCheckCircle = "fa fa-check-circle";
+    var faExclamationCircle = "fa fa-exclamation-circle";
+    var valueClass = "";
+    var iconClass = "";
+    var iconStyle = "";
+    if (metric.baseline === undefined) {
```

Comment:
though the case is empty, it ensures that the baseline is not undefined in the following else branches. Instead of removing I made the code more explicit by explicitly assigning null string to the 3 variables, as is expected in case of an undefined baseline for a metric, now the case is not empty anymore ;)